### ISSUE TITLE
Better Install documentation for D8 projects

### ISSUE THREAD

**Issue by mike-potter** [joachim-n/dorgflow#43]
I used to use dorgflow a lot for my D7 modules.  Now in D8 we are using composer for everything.  It is unclear from the README instructions how to properly install dorgflow when working on D8 projects.  The section about using `lakedrops/dorgflow` in the Packagist isn't clear.

What I'd like to do is simply install dorgflow globally via `composer global require vendor/dorgflow` but that doesn't seem to work.  Perhaps the patch for putting dorgflow into the vendor/bin folder will help.  But we also need the README updated with better instructions for installing.

**Comment by joachim-n** [comment]
>  It is unclear from the README instructions how to properly install dorgflow when working on D8 projects. The section about using `lakedrops/dorgflow` in the Packagist isn't clear.

I honestly have no idea, sorry.

I use dorgflow as mentioned in the readme: git clone to anywhere on your system, then symlink the executable into ~/bin which is in my $PATH.

> Perhaps the patch for putting dorgflow into the vendor/bin folder will help.

Worth a try, yes.

> But we also need the README updated with better instructions for installing.

I'm afraid that every time I try to do something with Composer, I end up either extremely frustrated with the software, or with its maintainers. So this is not something I'm going to get involved in -- it's not worth the stress.

I'll happily accept a PR for improvements though.

**Comment by mike-potter** [comment]
I was finally able to get it installed by ignoring the section about composer and just following the first instructions: 1) git clone dorgflow repo someplace, 2) composer install --no-dev, 3) symlink dorgflow to /usr/local/bin/dorgflow to make the command available globally without messing with my PATH.

I think the section about "Projects using composer..." is very misleading.  It talks about a composer package that doesn't seem to work or have the dorgflow binary.  I'd suggest maybe rewording that section of the README and making the first section clearer that it is needed whether doing D7 or D8.  Right now "Optional: For composer driven Drupal projects" reads more like "Drupal 8 people look here!" which it really isn't.

Sorry you have frustrations with Composer.  I've had really good luck with it and actually get frustrated now when I go back to D7 without it.  Would be sad if people stopped using dorgflow in D8 just because of this.

**Comment by frederickjh** [comment]
Ah, ha! So we are suppose to git clone it. Some how I thought we could use composer to install it. 1+ on needs better installation instructions.

**Comment by frederickjh** [comment]
@mike-potter instructions should be in the project's README.md

**Comment by joachim-n** [comment]
@mike-potter , @frederickjh could you review the pull request please?

**Comment by frederickjh** [comment]
I reviewed it @joachim-n but maybe @mike-potter can review the composer based workflow section as I have not worked with that.

**Comment by ccjjmartin** [comment]
I would like to suggest building a phar file like drupal-check: https://github.com/mglaman/drupal-check/wiki/Build-From-Source

**Comment by frederickjh** [comment]
I emailed @jurgenhaas to see if he would help us out on finishing this issue with some advice since he is the maintainer of the lakedrops/dorgflow composer package. It is hosted on a [private Gitlab](https://gitlab.lakedrops.com/composer/plugin/dorgflow) instance so that is why I emailed him. He does have a Github account. Hopefully we will hear from him soon.

**Comment by frederickjh** [comment]
@ccjjmartin I would like to suggest that you look and see if there is already an issue for that otherwise create a new one and stop hijacking a issue about documentation.

**Comment by jurgenhaas** [comment]
Thanks @frederickjh for reaching out. I have just updated the [README](https://gitlab.lakedrops.com/composer/plugin/dorgflow) of our composer plugin which by the way is a public project, just hosted on our own GitLab instance. The plugin itself is available from Packagist and can be used by anyone.

Hope the additional instructions in the readme are helpful. With the described workflow it is super simple to work on d.o issues at any time within a customer project and it takes only a few seconds to get the development environment ready to go, even if none of dorgflow had been available before. This is very encouraging for contributors to actually push back upstream to d.o when ever they come across any issues.

**Comment by frederickjh** [comment]
Here is a [link to the diff](https://gitlab.lakedrops.com/composer/plugin/dorgflow/-/commit/b2c29912f952a3bf76c21bd334c76fbbca51f572) of the commit for the updates to the documentation for `lakedrops/dorgflow`  that @jurgenhaas made.

**Comment by ccjjmartin** [comment]
I have now blocked @frederickjh

To the rest of the group, I created another issue to address the root cause of this issue which is composer installs are bound to result in "hard to install" experiences as stated by both the owner of this repository and the issue originator.  To continue the conversation and explore solutions in an open positive manner, see this issue: https://github.com/joachim-n/dorgflow/issues/53
